### ISSUE TITLE
[Question] Can this library be used for creating my own SMS/MMS app?

### ISSUE THREAD

**Issue by HubKing** [klinker41/android-smsmms#132]
I am sorry for asking here, but I could not find another way to ask a question. (I always thought it would be better if GitHub just adds "Discussions" other than "Issues".)

None of existing SMS/MMS apps (sadly, including yours) fulfills all my needs, so I want to create a new one. But the code for actually sending/receiving SMS/MMS is not my concern; It is probably the same for all SMS/MMS apps out there, and I do not want to spend months re-inventing the wheel. What I want to implement is mainly interface stuff and utility features.

So, I was looking for a library would do the daunting part of sending/receiving MMS messages, and I have found this library. What confuses me is that the description only mentions "sending", as if it is a library for non-SMS/MMS applications that occasionally want to send MMS.  But then, a paragraph below says that this library is used in Sliding Messaging Pro/EvolveSMS/Pluse which all seem to be SMS/MMS apps.

So, in short, if I want to create my own SMS/MMS app, can I use this library to send/receive (both) MMS including saving them into the Android's system database, so that I can focus on other features? Or is this library for apps who just want to send MMS occasionally?

**Comment by klinker41** [comment]
Yes, it is used in my apps. I think the sample has both sending and receiving examples to look at, I would start there.

**Comment by HubKing** [comment]
It seems that the example only sends MMS, not receives. The `MmsReceiver` class is empty.

**Comment by klinker41** [comment]
The class that it extends however is not empty, take a look there.
